spec/features/editing_content_settings/backdate_spec.rb

```diff
@@ -48,29 +41,38 @@ def and_i_enter_a_date_to_backdate_the_content_to
   end
 
   def and_i_click_save
-    @request = stub_publishing_api_put_content(@edition.content_id, {})
+    @set_request = stub_publishing_api_put_content(
+      @edition.content_id,
+      hash_including("first_published_at" => "2019-01-01T00:00:00.000+00:00"),
+    )
     click_on "Save"
   end
 
   def then_i_see_the_content_has_been_backdated
+    expect(@set_request).to have_been_requested
+
     expect(page).to have_content("1 January 2019")
     expect(page).to have_content(I18n.t!("documents.history.entry_types.backdated",
                                          date: "01 January 2019"))
-    expect(@request).to have_been_requested
   end
 
-  def given_there_is_a_document_with_a_second_edition
-    @edition = create(:edition, number: 2)
-  end
-
-  def then_i_cannot_see_a_backdate_section
-    expect(page).not_to have_content(
-      I18n.t!("documents.show.content_settings.backdate.title"),
+  def and_i_click_clear_backdate
+    @clear_request = stub_publishing_api_put_content(
+      @edition.content_id,
+      # hash_excluding doesn't seem to work here
+      ->(body) { JSON.parse(body).exclude?("first_published_at") },
```

Comment:
I had a look into it and I'm not sure why either. What's bizarre is that `hash_including` appears to be using the [WebMock matcher](https://github.com/bblimke/webmock/blob/a3c8de2550122d0998cb47292be0bbd2e3f203bf/lib/webmock/matchers/hash_excluding_matcher.rb#L11) (you can tell because of the different why they print on failure), whereas `hash_excluding` is using the RSpec one. I tried forcing the WebMock matcher but it didn't work either 😞.

Forcing both to WebMock and debugging it manually shows that, somehow, that `HashIncludingMatcher#==` gets the parsed JSON, while `HashExcludingMatcher#==` gets the raw string. I'm guessing something somewhere is doing a conditional parse based on the class. And it looks like I'm right...

https://github.com/bblimke/webmock/blob/a3c8de2550122d0998cb47292be0bbd2e3f203bf/lib/webmock/request_pattern.rb#L255

Reply:
Perhaps we can put a link to this comment...in the comment 😄.

Reply:
Amazing - great digging! I went down quite a rabbit hole myself experimenting with the different matchers for RSpec and WebMock and didn't come to a satisfying answer. Looks like you've nailed it though.

Also removed this now though.